docs/contextual-bandits/actions_contexts.md

```diff
@@ -0,0 +1,105 @@
+This gives the application a lot of control over the bandit: while the Contextual Bandit might pick any of the available actions in order to explore, it is possible to filter available actions based on particular requirements. 
+
+:::info An example of filtering actions
+Consider using the Contextual Bandit to select a product promotion for an e-commerce shop.
+Then there might be specific requirements such as not showing any maternity items to users that have not explicitly shared that they are pregnant.
+:::
+
+#### New actions
+
+Our Contextual Bandits automatically explore new actions as they are encountered, and over time will learn for which contexts these are most effective.
+
+### Contexts
+
+To learn a bandit policy that personalizes, you also need to provide context. 
+
+Generally, there are three types of context:
+1. Subject context: For example, the subjects's past behavior, demographic information, are they on a mobile device, etc.
+2. Action context: For example, the size of the discount, the type of product, the price, etc.
+3. Subject-action interaction context: For example, the brand affinity of the subject to the product (based on past purchases, user reviews, etc.)
+
+Note that the first of these is independent of the actions, while the other two are action dependent. 
+For convenience, you can supply the subject attributes once and they will be used across all actions, while
+separately, you can supply action specific attributes.
+Behind the scenes, we combine the two to create a single context per action that is used by the underlying model to select which action to pick.
+
+#### Subject attributes
+
+Subject attributes capture information about the subject that is relevant to the actions. 
+Generic attributes such as age (bucket), gender, device information can be helpful, but the most salient attributes are problem specific.
+
+#### Action attributes
+
+Action attributes capture information that is unique to a particular action. For example, the discount offered in a promotion, the price of a product, etc.
+Furthermore, you can also leverage action attributes to include information that is specific to the subject-action pair. For example, you can leverage an internal Machine Learning model that measures brand affinity.
+
+:::info What context attributes to use
+Selection of which attributes to include in the context is a bit of an art.
+
+In general, there are two types of attributes that will help the Contextual Bandit efficiently learn a strong policy:
+1. Attributes that provide a strong signal on personalization: For example, the brand affinity of the subject to the product, the price, etc.
+2. Attributes that predict the outcome of an action: For example, when optimizing for purchases, whether a user is a new or returning user might not affect which action is best, it can help reduce variance (similar to CUPED), helping the contextual bandit learn more efficiently
+:::
+
+:::warning Avoid high cardinality attributes
```

Comment:
I wonder if we should also state that we'll automatically cap cardinality if we're doing that yet (@giorgiomartini0 do we do this yet?)

Reply:
Yes [we do](https://github.com/Eppo-exp/eppo/blob/c5926af32a7632873beba7d6f4d143255c14e04e/packages/api/src/datawarehouse/db-aggregations/bandits/bandit-process-attributes.blocks.ts#L53): the cap is currently [set to 100](https://github.com/Eppo-exp/eppo/blob/c5926af32a7632873beba7d6f4d143255c14e04e/packages/api/src/regression/bandit-estimation.service.ts#L24). Attributes with more than 100 categories are dropped from the regression.